src/nav/navReducer.js

```diff
@@ -63,7 +63,10 @@ const rehydrate = (state, action) => {
   if (rehydratedState.users === undefined) {
     return getStateForRoute('loading');
   }
-
+  // Dont switch to main UI if sharing screen is on.
+  if (state.routes.find(route => route.routeName === 'sharing')) {
+    return state;
+  }
```

Comment:
Hmm interesting. Can you say more about why this is needed? What's the scenario in which it takes effect?

One thing that makes me curious here is that we don't have a similar condition in this spot for notifications, even though opening a notification can mean a very similar navigation at startup.

Reply:
Huh! That is peculiar. It leaves me still curious what's different between this and notifications that would make it needed here and not there.

I don't see this mentioned in the group PM thread with you and me and Chris. Probably #mobile-team would be the best place for any further debugging, in any case.

One possibility I wonder about is that perhaps it's related to how in a previous version of this PR there was a separate activity. Now that that's gone, the way the rest of the system works is more like it does for notifications.